src/VscPreviewExtension/media/buildTree.js

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+function buildTree(containerName, treeData) {
+  var i = 0;
+  var durationTime = 750;
+  var maxLabelLengthOfEachLevel = [0];
+  var port = document.body.childNodes[1].data;
+
+  var width = $(document).width();
+  var height = $(document).height();
+
+  var tree = d3.layout.tree()
+    .separation(function (a, b) { return a.parent = b.parent ? 1 : 2; });
+
+  var diagonal = d3.svg.diagonal()
+    .projection(function (d) { return [d.y, d.x]; });
+
+  // Call visit function to establish maxLabelLength of every level
+  visit(treeData, 0, function (d, level) {
+    var tmp = level;
+    // If the node has children , the text(node's name) will be displayed on the head of node , or on the right
+    if (!(d.children && d.children.length > 0))
+      tmp++;
+    // 'while' but not 'if' because it maybe skip the next level
+    while (maxLabelLengthOfEachLevel.length <= tmp)
+      maxLabelLengthOfEachLevel.push(minHorizontalSpace);
+    maxLabelLengthOfEachLevel[tmp] = Math.min(Math.max(maxLabelLengthOfEachLevel[tmp], getTokenName(d.name).length), 12);
```

Comment:
it is just a constant,maxHorizontalSpace

---
In reply to: [80409592](https://github.com/dotnet/docfx/pull/693#discussion_r80409592) [](ancestors = 80409592)